crates/bevy_ecs/src/schedule/schedule.rs

```diff
@@ -157,15 +160,29 @@ fn make_executor(kind: ExecutorKind) -> Box<dyn SystemExecutor> {
 }
 
 /// Chain systems into dependencies
-#[derive(PartialEq)]
-pub enum Chain {
-    /// Run nodes in order. If there are deferred parameters in preceeding systems a
-    /// [`apply_deferred`] will be added on the edge.
-    Yes,
-    /// Run nodes in order. This will not add [`apply_deferred`] between nodes.
-    YesIgnoreDeferred,
-    /// Nodes are allowed to run in any order.
-    No,
+#[derive(Default)]
+pub struct Chain(Option<BTreeMap<TypeId, Box<dyn Any>>>);
```

Comment:
This feels like a very complex and unidiomatic API for something relatively simple.

Reply:
I'm not entirely sure what that means. This is pretty much the minimal amount of information required to store edge options extensibly. If you have a better proposal, feel free to let me know. I've also added more docs to help explain this.